### ISSUE TITLE
Core: Deprecate `<table>` based layouts

### ISSUE THREAD

**Issue by Bukama** [primefaces/primefaces#9282]
### Describe the bug

When using the [SelectOneRadio](https://www.primefaces.org/showcase/ui/input/oneRadio.xhtml) with `layout` configuration set to `lineDirection`(wich is the default value!), then invalid HTML is rendered.

> The value of the for attribute of the label element must be the ID of a non-hidden form control

This is caused by the fact that with the rendering option `lineDirection` a table is rendered and the labels of the options point to the id of the table. But tables are not allowed to have a label.

![radioInput_tablerender](https://user-images.githubusercontent.com/4354359/194553141-94506c74-141b-48a6-95fd-bbe5a7c3803b.PNG)

### Reproducer

https://www.primefaces.org/showcase/ui/input/oneRadio.xhtml (Showcase Primefaces 12)

### Expected behavior

* Valid HTML is rendered, with the label pointing to the input field (hich could be problematic as it's hidden)
* No usage of layout tables, therefore change the default value to `responsive` or chaning the implementation of the `lineDirection` (and others)

### PrimeFaces edition

Community

### PrimeFaces version

11.0.0 at work (Showcase is using 12.0.0)

### Theme

Saga

### JSF implementation

All

### JSF version

2.2

### Java version

8

### Browser(s)

All

**Comment by jepsar** [comment]
I can have a look, but this might be a tough one. I'd like to know what PrimeTek has in mind for this. IMO this component could use an overhaul.

**Comment by Bukama** [comment]
> I can have a look, but this might be a tough one. I'd like to know what PrimeTek has in mind for this. IMO this component could use an overhaul.

Yes my first thought was also "remove the layout tables" at all, but this would be of course a larger overhaul, which should be discussed first by the leadership.

**Comment by melloware** [comment]
@Bukama bump ☝🏻

**Comment by Bukama** [comment]
Sry must have overseen this Github notification. We use HTML Nu Checker as our primary HTML validation tool.

**Comment by melloware** [comment]
OK well that NUHTML checker is notorious for false positives and bugs.  I believe our current code is correct if you manually review it or use a screen reader.

**Comment by melloware** [comment]
I will close for now because the HTML code is correct I think its a bug in the nuHTML checker.  If you find that the screen reader is not reading this component properly I will re-open.

**Comment by GedMarc** [comment]
Ok so - mine-as-well throw my cents in - had to actually look at this as well cause something is not right here 

This is HTML401 structure - not many browsers/readers even support it anymore, some have backwards compatibility but most just removed the table-structured layout and used a CSS property for it - https://www.w3schools.com/cssref/pr_tab_table-layout.php
nuHTML checker is not the tool to use AT ALL :) It should have flagged this structure as invalid for HTML5 - https://stackoverflow.com/questions/15642375/avoiding-table-for-html-5-layout (9 years old..)

There has to be a limit for how far back backwards compatibility goes, or how far back support for it should go. 

@Bukama time to have that talk with your leadership

**Comment by melloware** [comment]
@GedMarc that is a great point I wonder if we should open a ticket that is replace all TABLE layouts with DIV+CSS layouts in components still using TABLE's?  @christophs78 ? @jepsar ?

**Comment by jepsar** [comment]
+1 we could also deprecate the old custom content in favor of the facet solution.

**Comment by Bukama** [comment]
As written in my opening post I also voted for changing the default value to `responsive` which is a rendering without the layout table (which as stated is the old way and also non correct semantik HTML).
We changed to this after we realized that the default value of the SelectOneRadio generates layout tables.

So you could change the default value in PF 13 and mark layout table as deprecated for removal (in maybe 14). I think that's a good way to go and gives the users at least half a year to change it.

**Comment by christophs78** [comment]
Yep. HTML-tables used for layout should be history since at least a decade.
Adding a note to migration-guide and ideally provide some backward-compatibility.
+1 for changing default-value for layout-attribute

**Comment by melloware** [comment]
Ok I am on it. Will add migration guide, deprecation warnings, and default to responsive layouts.

**Comment by melloware** [comment]
Here is what I put in the migration guide...

  * `SelectOneRadio` tabular layout in favor of responsive layout
  * `SelectManyCheckbox` tabular layout in favor of responsive layout
  * `DataGrid` tabular layout in favor of responsive layout `layout="grid"` now the default
  * `PanelGrid` tabular layout in favor of responsive layout `layout="grid"` now the default. To fall back to legacy behavior set `layout="tabular" columns="0"`

**Comment by tandraschko** [comment]
I just tested the new PF version within our applications and `p:selectOneMenu layout="pageDirection"` is now broken
it was line based before (each radio in a new line), now its column based without table

**Comment by tandraschko** [comment]
@melloware can you fix this?

**Comment by melloware** [comment]
@tandraschko yep looking into it!
